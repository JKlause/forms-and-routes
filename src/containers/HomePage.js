import React, { Component } from 'react';
import RandomImage from '../components/RandomImage';
import CharacterSearchForm from '../components/CharacterSearchForm';
import { callApi } from '../service/callApi';

export default class HomePage extends Component {

  state = {
    photoUrl: '',
    name: '',
    searchQuery: ''
  }

  componentDidMount() {
    callApi('random')
      .then(([{ photoUrl, name }]) => {
        this.setState({ photoUrl, name });
      });
  }

  handleSubmit = event => {
    event.preventDefault();

    callApi(this.state.searchQuery)
      .then(([{ photoUrl, name }]) => {
        this.setState({ photoUrl, name });
      });
  }

  handleRandomGet = () => {
    callApi('random')
      .then(([{ photoUrl, name }]) => {
        this.setState({ photoUrl, name });
      });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {

    const randomImageProps = {
      photoUrl: this.state.photoUrl,
      name: this.state.name,
      handleRandomGet: this.handleRandomGet
    };

    const CharacterSearchFormProps = {
      searchQuery: this.state.searchQuery,
      handleSubmit: this.handleSubmit,
      handleChange: this.handleChange
    };
    
    return (
      <div>
        <RandomImage {...randomImageProps}/>
        <CharacterSearchForm {...CharacterSearchFormProps}/>
      </div>
    );
  }
}
