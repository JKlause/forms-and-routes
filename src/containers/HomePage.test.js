import React from 'react';
import { shallow } from 'enzyme';
import HomePage from './HomePage';



describe('HomePage container', () => {

  it('renders HomePage component', () => {

    // eslint-disable-next-line no-undef
    global.fetch = jest.fn(() => Promise.resolve());
    const wrapper = shallow(<HomePage history={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('handleRandomGet changes state.photoUrl and state.name', () => {
 
  });

  it('handleChange changes state of target', () => {
    
  });
});
