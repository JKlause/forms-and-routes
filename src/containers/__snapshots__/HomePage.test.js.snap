// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage container renders HomePage component 1`] = `
<img
  src="https://media.giphy.com/media/w461tXo03sQP6/giphy.gif"
/>
`;
